cpp/template/cmake/thirdparty/fetch_rapids.cmake

```diff
@@ -0,0 +1,20 @@
+# =============================================================================
+# Copyright (c) 2023, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+
+set(RAPIDS_VERSION "23.04")
+
+if(NOT EXISTS ${CMAKE_CURRENT_BINARY_DIR}/FAISS_RAPIDS.cmake)
+    file(DOWNLOAD https://raw.githubusercontent.com/rapidsai/rapids-cmake/branch-${RAPIDS_VERSION}/RAPIDS.cmake
+            ${CMAKE_CURRENT_BINARY_DIR}/FAISS_RAPIDS.cmake)
```

Comment:
`FAISS_RAPIDS`?

Reply:
Ah, yep. Can you tell I copied this from the FAISS PoC?

Reply:
But doesn't this file not exist in `rapids-cmake` repo? What is happening here exactly?

Reply:
This is the same thing that's in all of our `fetch_rapids.cmake` files (in all of the RAPIDS projects). It's just bringing down the  rapids-cmake files and including them in the project. You can really change `FAISS_RAPIDS` to be whatver you want- it just pulls down a cmake file and renames it to that.